File: src/main/java/com/demo/react/server/templates/Html.java
package com.demo.react.server.templates;

import org.springframework.stereotype.Component;

import java.io.InputStream;
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

@Component
public class Html {

    private String left;
    private String right;

    public Html() {
        InputStream in = getClass().getClassLoader().getResourceAsStream("static/index.html");
        Scanner s = new Scanner(in).useDelimiter("\\A");
        String index = s.hasNext() ? s.next() : "";
        String regex = "[\\s\\S]+<div id=\"root\">";
        Pattern p = Pattern.compile(regex);
        Matcher matcher = p.matcher(index);
        matcher.find();
        left = matcher.group(0);
        String[] split = index.split(regex);
        right = split[1];
    }

    public String getIndex(String element) {
        return left + element + right;
    }

}

File: src/main/kotlin/com/demo/react/server/templates/Html.kt
package com.demo.react.server.templates

import com.demo.react.server.utils.Utils.read
import org.springframework.stereotype.Component

@Component
class Html {

    private val indexGame = read("static/index.game.html")
    private val indexApp = read("static/index.app.html")
    private val regex = Regex("[\\s\\S]+<div id=\"root\">")

    fun getSimpleIndex(element: String) = html(indexGame, element)

    fun getMaterialUIindex(element: String, css: String) = html(indexApp, "$element</div><style id=\"jss-server-side\">$css</style>")

    private fun html(index: String, body: String) = "${regex.find(index)?.value}$body${regex.split(index)[1]}"

}
